app/models/reports/graphs.rb

```diff
@@ -18,5 +18,14 @@ def self.order_count_by_month
       order_count.map { |k, v| results[k] = v.count }
       Hash[results.map { |k, v| [k.to_date, v] }]
     end
+
+    def self.item_count_by_month
+      results = {}
+
+      order_details = OrderDetail.joins(:order).where(orders: { status: 6 })
+                                 .group_by { |m| m.created_at.beginning_of_month }
+      order_details.map { |datetime, od| [results[datetime] = od.map(&:quantity).sum] }
```

Comment:
Looks like you just need `each` here, not `map`
Also, you won't need to return the array that you are currently returning
And `results[datetime]` can be `results[datetime.to_date]` so you don't have to change the results in the next line (so the next line can just be `results` instead of building a new Hash)